File: app/src/main/java/ru/eva/oriokslive/activities/news/NewsActivity.java
package ru.eva.oriokslive.activities.news;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import ru.eva.oriokslive.R;

public class NewsActivity extends AppCompatActivity implements ContractNewsActivity.View {

    private ContractNewsActivity.Presenter mPresenter;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news);
        mPresenter = new PresenterNewsActivity(this);

        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        Intent intent = getIntent();
        WebView webView = findViewById(R.id.webView);
        webView.loadUrl(intent.getStringExtra("url"));
        webView.setWebViewClient(new WebViewClient() {
            @Override
            public void onPageFinished(WebView view, String url) {
                getSupportActionBar().setDisplayHomeAsUpEnabled(true);
                getSupportActionBar().setTitle(view.getTitle());
                getSupportActionBar().setSubtitle(view.getUrl());
            }});
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        finish();
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/ru/eva/oriokslive/ui/activity/news/NewsActivity.kt
package ru.eva.oriokslive.ui.activity.news

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.MenuItem
import android.webkit.WebView
import android.webkit.WebViewClient
import ru.eva.oriokslive.databinding.ActivityNewsBinding
import ru.eva.oriokslive.ui.base.BaseActivity
import ru.eva.oriokslive.ui.fragment.news.NewsFragment.Companion.EXTRA_URL

class NewsActivity : BaseActivity<ActivityNewsBinding>() {
    override val bindingInflater: (LayoutInflater) -> ActivityNewsBinding =
        ActivityNewsBinding::inflate

    private val url: String by lazy { intent.getStringExtra(EXTRA_URL) ?: "" }

    @SuppressLint("SetJavaScriptEnabled")
    override fun setupUI() {
        setSupportActionBar(binding.toolbar)
        binding.webView.webViewClient = getClient()
        binding.webView.settings.javaScriptEnabled = true
        binding.webView.loadUrl(url)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        finish()
        return true
    }

    private fun getClient(): WebViewClient = object : WebViewClient() {
        override fun onPageFinished(view: WebView, url: String) {
            supportActionBar?.apply {
                setDisplayHomeAsUpEnabled(true)
                title = view.title
                subtitle = view.url
            }
        }
    }
}
